utils/src/spammer.rs

```
@@ -165,91 +187,120 @@ impl Spammer {
+            // Verify the nonce for gaps
+            // TODO: probably this should run as a separate task
+            let on_chain_nonce = self.get_latest_nonce(address).await?;
+            // If the span between the on-chain nonce and the one we are about to send
+            // is too big, then probably there is a gap that doesn't allow the
+            // on-chain nonce too advance.
+            let nonce_span = nonce.saturating_sub(on_chain_nonce);
+            if nonce_span > self.max_rate {
+                debug!("Current nonce={nonce}, on-chain nonce={on_chain_nonce}. Sending 10 txs");
+                let batch_entries = self.build_batch_entries(10, on_chain_nonce).await?;
+                if let Some(results) = self.send_raw_batch(&batch_entries).await? {
+                    if results.len() != batch_entries.len() {
+                        return Err(eyre::eyre!(
+                            "Batch response count {} does not match request count {}",
+                            results.len(),
+                            batch_entries.len()
+                        ));
+                    }
 
-            for _ in 0..self.max_rate {
-                // Check exit conditions before creating each transaction.
-                if (self.max_num_txs > 0 && txs_sent_total >= self.max_num_txs)
-                    || (self.max_time > 0 && start_time.elapsed().as_secs() >= self.max_time)
-                {
-                    break;
+                    // Report individual results.
+                    for ((_, tx_bytes_len), result) in batch_entries.into_iter().zip(results) {
+                        let mapped_result = result.map(|_| tx_bytes_len);
+                        result_sender.send(mapped_result).await?;
+                    }
+                } else {
+                    debug!("Batch eth_sendRawTransaction timed out; skipping this tick");
+                    let _ = report_sender.send(interval_start).await;
```

Why are we discarding the error?
And not something like:

```suggestion
                    report_sender.send(interval_start).await?;
```